pkg/networkmanager/interfaces.js

```diff
@@ -3256,6 +3243,32 @@ function apply_master_slave(choices, model, apply_master, master_connection, mas
         return cockpit.all(deferreds.get());
     }
 
+    if (!master_connection) {
+        var active_settings = [ ];
+
+        if (master_settings.bond &&
+            master_settings.bond.options &&
+            master_settings.bond.options.primary) {
+            var iface = model.find_interface(master_settings.bond.options.primary);
+            if (iface && iface.MainConnection)
+                active_settings.push(iface.MainConnection.Settings);
+        } else {
+            choices.find('input[data-iface]').map(function (i, elt) {
+                if ($(elt).prop('checked')) {
+                    var iface = model.find_interface($(elt).attr("data-iface"));
+                    if (iface.Device && iface.Device.ActiveConnection && iface.Device.ActiveConnection.Connection) {
+                        active_settings.push(iface.Device.ActiveConnection.Connection.Settings);
+                    }
+                }
+            });
+        }
+
+        if (active_settings.length == 1) {
+            master_settings.ipv4 = $.extend(true, { }, active_settings[0].ipv4);
+            master_settings.ipv6 = $.extend(true, { }, active_settings[0].ipv6);
```

Comment:
`$.extend(true, {}, foo)` is a deep copy, while `cockpit.extend({}, foo)` is shallow, no?